test/unit/test_http.py

```diff
@@ -45,15 +42,32 @@ def return_okay(url, request):
 @urlmatch(netloc='localhost:5001', path=r'.*/fail')
 def return_fail(url, request):
     """Defines an endpoint for failed http requests.
-
     This endpoint will listen at http://localhost:5001/*/fail for incoming
     requests and will always respond with a 500 status code and a Message of
     "fail".
+    Keyword arguments:
+    url -- the url of the incoming request
+    request -- the request that is being responded to
+    """
+    return {
+        'status_code': 500,
+        'content': 'fail'.encode('utf-8'),
+    }
+
 
+@urlmatch(netloc='localhost:5001', path=r'.*/fail')
```

Comment:
The problem is that this matches `/fail`, but your test expects `/timeout`. Please change this to the later and the tests will likely pass. The codestyle issue will disappear if you rebase onto current the current `py-ipfs-http-client` commit, so please do that as well as there are merge-conflicts atm.